src/Hive.Core/sia380/sia380.py

```diff
@@ -269,58 +278,131 @@ def main(room_properties, floor_area, T_e, setpoints_ub, setpoints_lb, surface_a
+        eta_g_lb_heating_no_hr = calc_eta_g(gamma_lb_heating_no_hr, tau, False)
+        eta_g_ub_heating_no_hr = calc_eta_g(gamma_ub_heating_no_hr, tau, False)
+        eta_g_lb_cooling_no_hr = calc_eta_g(gamma_lb_cooling_no_hr, tau, True)
+        eta_g_ub_cooling_no_hr = calc_eta_g(gamma_ub_cooling_no_hr, tau, True)
 
         # heating demand (Heizwärmebedarf), Q_H
-        demand_ub = Q_T_ub + Q_V_ub - eta_g_ub * (Q_i[month] + Q_s[month])
-        demand_lb = Q_T_lb + Q_V_lb - eta_g_lb * (Q_i[month] + Q_s[month])
-        eta_g = 0.0
-        if abs(demand_lb) < abs(demand_ub):
-            Q_T[month] = Q_T_lb
-            QT_opaque[month] = sum(QT_op_per_srf_this_month_lb)
-            QT_transparent[month] = sum(QT_tr_per_srf_this_month_lb)
-            Q_V[month] = Q_V_lb
-            eta_g = eta_g_lb
-        else:
-            Q_T[month] = Q_T_ub
-            QT_opaque[month] = sum(QT_op_per_srf_this_month_ub)
-            QT_transparent[month] = sum(QT_tr_per_srf_this_month_ub)
-            Q_V[month] = Q_V_ub
-            eta_g = eta_g_ub
-
-        Q_i_eta_g[month] = Q_i[month] * eta_g
-        Q_s_eta_g[month] = Q_s[month] * eta_g
-        demand = Q_T[month] + Q_V[month] - (Q_i_eta_g[month] + Q_s_eta_g[month])
-        if demand > 0:
-            Q_Heat[month] = demand
+        Q_H_ub = Q_T_ub + Q_V_ub_heating - eta_g_ub_heating * (Q_i[month] + Q_s[month])
+        Q_H_lb = Q_T_lb + Q_V_lb_heating - eta_g_lb_heating * (Q_i[month] + Q_s[month])
+        Q_H_ub_no_hr = Q_T_ub + Q_V_ub_heating_no_heat_recovery - eta_g_ub_heating_no_hr * (Q_i[month] + Q_s[month])
+        Q_H_lb_no_hr = Q_T_lb + Q_V_lb_heating_no_heat_recovery - eta_g_lb_heating_no_hr * (Q_i[month] + Q_s[month])
+        if Q_H_ub < 0:
+            Q_H_ub = 0
+        if Q_H_lb < 0:
+            Q_H_lb = 0
+        if Q_H_ub_no_hr < 0:
+            Q_H_ub_no_hr = 0
+        if Q_H_lb_no_hr < 0:
+            Q_H_lb_no_hr = 0
+
+        # cooling demand (Kältebedarf), Q_K
+        Q_K_lb = Q_i[month] + Q_s[month] - eta_g_lb_cooling * (Q_T_lb + Q_V_lb_cooling)
+        Q_K_ub = Q_i[month] + Q_s[month] - eta_g_ub_cooling * (Q_T_ub + Q_V_ub_cooling)
+        Q_K_lb_no_hr = Q_i[month] + Q_s[month] - eta_g_lb_cooling_no_hr * (Q_T_lb + Q_V_lb_cooling_no_heat_recovery)
+        Q_K_ub_no_hr = Q_i[month] + Q_s[month] - eta_g_ub_cooling_no_hr * (Q_T_ub + Q_V_ub_cooling_no_heat_recovery)
+        if Q_K_lb < 0:
+            Q_K_lb = 0
+        if Q_K_ub < 0:
+            Q_K_ub = 0
+        if Q_K_lb_no_hr < 0:
+            Q_K_lb_no_hr = 0
+        if Q_K_ub_no_hr < 0:
+            Q_K_ub_no_hr = 0
+
+        Q_H_list = [Q_H_lb, Q_H_ub, Q_H_lb_no_hr, Q_H_ub_no_hr]
+        Q_K_list = [Q_K_lb, Q_K_ub, Q_K_lb_no_hr, Q_K_ub_no_hr]
+        Q_H = min(Q_H_list)   # take smaller value of both comfort set points
+        Q_K = min(Q_K_list) * -1    # make cooling negative 
+        Q_H_index = Q_H_list.index(min(Q_H_list))
```

Comment:
You have already computed `min(Q_H_list)`... just use `Q_H_list.index(Q_H)`. But... also, you could have a function:

```
def min_and_index(*seq):
    min_value = min(seq)
    index = seq.index(min_value)
    return min_value, index

Q_H = min_and_index(Q_H_lb, Q_H_ub, Q_H_lb_no_hr, Q_H_ub_no_hr)
```

Then you don't need to create `Q_H_list` (yeah. it's hard to come up with a good name for it. that's often an indication for looking for something else.

Another thing to consider: Maybe `Q_H` could be it's own type?

Reply:
what do you mean, `Q_H` could be its own type?

Reply:
@daren-thomas answered me: Q_H could be an instance of a class (what to call Q?) that does these calculations and has lb, ub etc. properties. it's not important...

will do in maybe some possible future in some possible multiverse